igvc_navigation/src/path_follower/SmoothControl.cpp

```diff
@@ -35,81 +31,57 @@ void SmoothControl::getTrajectory(igvc_msgs::velocity_pair& vel, nav_msgs::Path&
     pose.pose.position.x = end_pos[0];
     pose.pose.position.y = end_pos[1];
     trajectory.poses.push_back(pose);
-    // std::cout << std::endl;
     cur_pos = end_pos;
   }
-  // std::cout << "\n end" << std::endl;
 }
 
-void SmoothControl::getAction(Eigen::Vector3d& result, Eigen::Vector3d cur_pos, Eigen::Vector3d target)
+void SmoothControl::getAction(Eigen::Vector3d& result, Eigen::Vector3d cur_pos,
```

Comment:
Small detail, but is there a reason why we don't combine position and heading into one 5d vector?
It is a little confusing as to which position vector it corresponds to (I'm assuming it is the current position).

Reply:
so cur_pos (current pose, _not_ current position) and egocentric_heading are a bit different. cur_pose holds odometry information relative to odom, (x,y,yaw). egocentric_heading holds two angles that are essential to calculating a trajectory but are not relative to odom, they're instead relative to the line of sight from the robot to the target pose.

If you would like a more detailed explanation, the paper does a good job explaining what these parameters mean

Reply:
That makes sense. Thanks for clearing that up